### DISCUSSION TITLE
There are NaNs in y_pred (Pipeline.backtest)

### DISCUSSION THREAD

**Discussion by Darya031** [etna-team/etna#472] [Q&A]
why there is mistake  "There are NaNs in y_pred" while I'm doing Pipeline.backtest with NBeatsGenericModel model 
(timeseries don't contain NaN values)

here is my code:
```HORIZON = 30
metrics = [SMAPE(), MAPE(), MAE()]
model_nbeats_generic = NBeatsGenericModel(
        input_size= HORIZON*2,
        output_size=HORIZON,
        loss="mape",
        stacks=30,
        layers=4,
        layer_size=256,
        trainer_params=dict(max_epochs=1000),
        lr=1e-3,
    )

pipeline_nbeats_generic = Pipeline(
        model=model_nbeats_generic,
        horizon=HORIZON,
        transforms=[])

metrics_nbeats_generic, forecast_nbeats_generic, _ = pipeline_nbeats_generic.backtest(
    ts, metrics=metrics, n_folds=3, n_jobs=1)```

**Comment by d-a-bunin** [comment]
Hi! How long is the timeseries (number of timestamps) and how long is `HORIZON`?

**Comment by Darya031** [comment]
The horizon is 30 days and there are 731 row in timeseries

**Comment by d-a-bunin** [comment]
- You could run `get_historical_forecasts` instead of `backtest`. This way you can inspect the backtest predictions without calculating metrics. Are there any NaNs?
- You could run `fit`, `forecast` and check the forecasts of the network. Are there any NaNs?

**Reply by Darya031** [reply]
There are NaNs if I'm doing get_historical_forecasts instead of backtest. There are also NaNs in predictions when I run fit, forecast and check the forecasts of the network

**Reply by d-a-bunin** [reply]
Let's dive into making `fit`, `forecast`. 
- Are NaNs in all prediction timestamps or only in some of them? 
- Are there any missing values in target history?
- Are there any missing values in exogenous variables (if there are any)?
